core/src/main/java/org/elasticsearch/index/query/MatchAllQueryParser.java

```diff
@@ -41,22 +41,24 @@ public MatchAllQueryParser() {
 
     @Override
     public MatchAllQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException {
-        MatchAllQueryBuilder queryBuilder = new MatchAllQueryBuilder();
         XContentParser parser = parseContext.parser();
 
         String currentFieldName = null;
         XContentParser.Token token;
+        float boost = 1.0f;
         while (((token = parser.nextToken()) != XContentParser.Token.END_OBJECT && token != XContentParser.Token.END_ARRAY)) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = parser.currentName();
             } else if (token.isValue()) {
                 if ("boost".equals(currentFieldName)) {
-                    queryBuilder.boost(parser.floatValue());
+                    boost = parser.floatValue();
                 } else {
                     throw new QueryParsingException(parseContext, "[match_all] query does not support [" + currentFieldName + "]");
                 }
             }
         }
+        MatchAllQueryBuilder queryBuilder = new MatchAllQueryBuilder();
```

Comment:
It was okay but in most of the other parsers we follow the pattern of first collecting all the values we need and then constructing the new query builder at the end of the method. I think that way it's easier to read and you have all the code that constructs the return query in one place.